File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        
        while (l < r) {
            int mid = l + (r - l) / 2;
            
            if (nums[mid] > nums[r]) {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        int min = l; 
        
        if(target >= nums[min] && target <= nums[nums.length-1] ){
            l = min;
            r = nums.length-1;
            
        }
        else{
            l = 0;
            r = min;
        }
        while( l <= r){
                int mid = l + (r-l)/2;
                if(nums[mid]==target){
                    return mid;
                }
                else if(nums[mid] > target){
                    r = mid-1;
                }
                else{
                    l = mid + 1;
                }
            }
        return -1;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l , r =0, len(nums)-1
        low = float('inf')

        while l < r:
            mid = l + (r-l)//2
            low = min(low,nums[mid])
            if nums[mid] > nums[r]:
                l = mid+1
            else:
                r = mid-1
        return min(low,nums[l])


        